components/salsa-macros/src/query_group.rs

```diff
@@ -499,10 +509,27 @@ pub(crate) fn query_group(args: TokenStream, input: TokenStream) -> TokenStream
             };
             let invoke = query.invoke_tt();
 
+            let update = if let Some(update) = &query.update {
+                quote! {
+                    fn update<MP: salsa::plumbing::MemoizationPolicy<Self>>(
+                        db: &<Self as salsa::QueryDb<'_>>::DynDb,
+                        #key_pattern: <Self as salsa::Query>::Key,
+                        value: &mut <Self as salsa::Query>::Value,
+                    ) -> bool {
+                        #update(db, value, #(#key_names),*)
```

Comment:
I'm curious why the keys come *after* the value -- that somewhat surprised me. It doesn't really matter, of course.

Reply:
I think it's my habit to put `&mut` parameters at the front, `strcpy`-style. I agree there is a tension vs `key`-`value`, and the framework internals (eg `QueryFunction`) have `key` then `value`.